all_repos/push/gitlab_pull_request.py

```diff
@@ -33,8 +77,7 @@ def push(settings: Settings, branch_name: str) -> None:
     }
 
     remote_url = git.remote('.')
-    _, _, repo_slug = remote_url.rpartition(':')
-    repo_slug = _strip_trailing_dot_git(repo_slug)
+    repo_slug = _extract_gitlab_project_path(remote_url)
```

Comment:
all-repos intentionally only supports ssh remotes -- if you want support for some other format please keep those in your own `source`es / `push`es

Reply:
by ssh I mean explicitly *only* `git@host:repo_slug` as is already done

Reply:
But this syntax does not support specifying port (which is possible installation of a self hosted gitlab)

Reply:
In my understanding, git insteadof won't solve the problem here as it's a parsing problem.

Anyway, I get that you don't want to support anything else then scp format. But that don't get verified and it's very hard to understand, maybe you want to improve the error messages.

What was also frustrating is that even the PR creation failed, the execution continued and left me with a lot of repos with branches without PRs. And even if I would want to run it again with fixed module it won't work as a branch with the same name already exists in the remote.

I don't know it it's something that you want to fix.

So, you want me to drop that commit and leave only the other fix?

Reply:
yes and generally a PR should only do one thing anyway so lumping these together was already a bad idea